lib/internal/net.js

```diff
@@ -9,3 +9,9 @@ function isLegalPort(port) {
     return false;
   return +port === (port >>> 0) && port >= 0 && port <= 0xFFFF;
 }
+
+
+function assertPort(port) {
+  if (typeof port !== 'undefined' && !isLegalPort(port))
```

Comment:
Sorry, just getting back to this one. Since https://github.com/nodejs/node/issues/5733 landed, `isLegalPort(undefined)` now returns false. @dirceu, not certain if that changes any of the details that you're doing here. Can you take another look?